### DISCUSSION TITLE
[@types/ramda] new `prop`: wrong type work with map

### DISCUSSION THREAD

**Discussion by suienzan** [DefinitelyTyped/DefinitelyTyped#62985] [Issues with a @types package]
The new change in [#62789] (https://github.com/DefinitelyTyped/DefinitelyTyped/pull/62789) seems to break something.

```typescript
import { prop } from 'ramda';

const objArray = [{ foo: 'bar' }];
const barArray = objArray.map(prop('foo'));
```

The `barArray` will be indicated to `undefined[]` while it should be `string[]`.

[a comment by an automated account is not shown]

**Comment by Nicholaiii** [comment]
@anion155 any ideas?

**Reply by anion155** [reply]
I'll need couple hours to investigate, what can be done. But for now, if we stop support for `R.prop<'x', number>('x')(obj)` and `R.prop<'x'>('x')<number>(obj)`, it would be resolved

**Reply by anion155** [reply]
```
export function prop<P extends keyof never>(p: P): {
    // <T>(value: Record<P, T>): T;
    <T>(value: T): Prop<T, P>;
}
```

**Reply by anion155** [reply]
@Nicholaiii Maybe it would be better to revert current changes and make a `breaking changes update` without support for old syntax if any?

**Comment by anion155** [comment]
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/63184
